Arin/8988 data retention change modal

arinkulshi-skylight opened the pull request:
# FRONTEND PULL REQUEST

https://app.zenhub.com/workspaces/simplereport-2025-and-onwards-605b43a6ceb92c000f86279a/issues/gh/cdcgov/prime-simplereport/8988

## Related Issue

This Issue is part of the data retention policy

## Testing 

You can run this locally and see the modal
You can run the tests locally

## Changes Proposed

Creates a modal to outline the new data retention policy
Modal will not show if the user checks "Got it. I don't need to see it again" 

## Screenshot

<img width="748" height="429" alt="Screenshot 2025-07-24 at 8 29 55 AM" src="https://github.com/user-attachments/assets/ce799c3a-a8da-4436-b1f2-4a7deb455c56" />


<!---
## Checklist for Author and Reviewer
### Accessibility
- [ ] Any large changes have been run through Deque manual testing
- [ ] All changes have run through the Deque automated testing

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [ ] Any content changes have been approved by content team

### Support
- [ ] Any changes that might generate new support requests have been flagged to the support team
- [ ] Any changes to support infrastructure have been demo'd to support team

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### Documentation
- [ ] Any changes to the startup configuration have been documented in the README
-->

mpbrown reviewed (approved):
Tested locally and LGTM! Nice work!

sarahtress commented:
@arinkulshi-skylight thanks for this! 3 smaller requests:

- can the buttons be aligned to the left? right now they're right aligned
- is it possible to get rid of the blue focus state border around the modal? it's sort of distracting and not expected
- can you reduce the spacing between the "learn more" text and the icon to be closer to the spacing in the designs? the spacing right now is too large

1 larger request:
- looks like the modal appears on the log in screen, when I would only want it to appear AFTER somebody has logged into a facility. is it possible to change the functionality so it only shows after somebody has selected a facility?

arinkulshi-skylight commented:
> @arinkulshi-skylight thanks for this! 3 smaller requests:
> 
> * can the buttons be aligned to the left? right now they're right aligned
> * is it possible to get rid of the blue focus state border around the modal? it's sort of distracting and not expected
> * can you reduce the spacing between the "learn more" text and the icon to be closer to the spacing in the designs? the spacing right now is too large
> 
> 1 larger request:
> 
> * looks like the modal appears on the log in screen, when I would only want it to appear AFTER somebody has logged into a facility. is it possible to change the functionality so it only shows after somebody has selected a facility?

Awesome I have aligned the buttons to the left, removed the focus border and reduced the spacing, I have moved the Modal to the page. When users access the dashboard page they will see the modal. I have deployed the changes into dev3. I have also included David's changes as a part of this ticket we will merge both in one ticket.

sarahtress commented:
> @arinkulshi-skylight I just realized the modal only shows on the dashboard which only facility admin users have access to. If this is supposed to show for all users, we probably need to either move it to the Report Tests page or show it on both Report Tests and Dashboard.
> 
> For example here is the default page for entry only users <img alt="image" width="1285" height="552" src="https://private-user-images.githubusercontent.com/23287037/473563262-7ee1684d-581b-49a1-83fc-5d62d2afed52.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQwNzI2MzQsIm5iZiI6MTc1NDA3MjMzNCwicGF0aCI6Ii8yMzI4NzAzNy80NzM1NjMyNjItN2VlMTY4NGQtNTgxYi00OWExLTgzZmMtNWQ2MmQyYWZlZDUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODAxVDE4MTg1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwMmRmMGRhNDdkZTQxOGFmZDgzMmJkMTYxMTFlMWFjYjUxMDA2NTUyOTE5NGQ2NDVhNmJkNTc4YzMwMTkxOWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hd1GMgKV0sp7hZmOouxRou9l5FGygNOQe3dsAeamrXc">
> 
> And the default page for basic users <img alt="image" width="1281" height="527" src="https://private-user-images.githubusercontent.com/23287037/473563143-3b60a446-46f7-4945-babf-4a3dfe1081d9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQwNzI2MzQsIm5iZiI6MTc1NDA3MjMzNCwicGF0aCI6Ii8yMzI4NzAzNy80NzM1NjMxNDMtM2I2MGE0NDYtNDZmNy00OTQ1LWJhYmYtNGEzZGZlMTA4MWQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODAxVDE4MTg1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEzNGQ4YjkzM2Q5YTk3ZDY0ZWQ1OTViY2UwZDUxNGMxYzU4MDkzODM0YTdiMTc5MzBmNDI0YzVlZTI2YjVlMTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DUykLvzJnFoQPTeo6U0oIQkaPS6kwBcEnhLmY1PEaqU">

yes, actually it should show up once when the user logs in no matter what page they're on. if they DON'T click the "got it" checkbox, it shouldn't show up again until they are timed out or log out/log back in. but right now, every time they click on the dashboard page it would appear, which is not ideal. let me know if this isn't clear

mpbrown reviewed (dismissed):
The update on dev3 looks good, confirmed that the modal shows up for other pages after doing a force refresh to test it. Thanks for updating that!

mehansen commented:
<img width="1436" height="1082" alt="Screenshot 2025-08-04 at 11 40 18 AM" src="https://github.com/user-attachments/assets/7ea27274-ebe0-4d08-9542-5c452139cdad" />
I think we generally don't worry about Safari, but just as an FYI the modal looks a little funky in Safari

mehansen reviewed (dismissed):
lgtm! left a couple questions/comments but don't think they should block esp since we want to get this in tonight

arinkulshi-skylight commented on frontend/src/app/ReportingApp.test.tsx:
```diff
@@ -249,7 +249,9 @@ describe("App", () => {
     expect(trainingWelcome).toBeInTheDocument();
 
-    await user.click(screen.getByText("Got it", { exact: false }));
+    await user.click(
+      screen.getByText(/got it,? thanks/i, { selector: "button" })
+    );
```
haha yeah it was bugging me but I guess it works for now

arinkulshi-skylight commented on frontend/src/app/commonComponents/DataRetentionModal.tsx:
```diff
@@ -0,0 +1,127 @@
+  const [dontShowAgain, setDontShowAgain] = useState(false);
+  const appInsights = getAppInsights();
+
+  if (window.Cypress) {
```
We do not show the modal when running cypress

arinkulshi-skylight replied:
We do not show the modal when running the cypress
